src/tox/tox_env/python/pip/pip_install.py

```diff
@@ -135,18 +135,14 @@ def _install_requirement_file(self, arguments: PythonDeps, section: str, of_type
                         raise Recreate(f"requirements removed: {' '.join(missing_requirement)}")
                     if old.get("constraint_options") != constraint_options:
                         raise Recreate(
```

Comment:
```
msg =  f"constraint options changed: old={old.get('constraint_options')} new={constraint_options}"
raise Recreate(msg)
```
would save you a line here 🤔

Reply:
i ended up making it a little cleaner by assigning `old_constraint_options = old.get("constraint_options")`